### DISCUSSION TITLE
No softmax in densenet

### DISCUSSION THREAD

**Discussion by grudloff** [Project-MONAI/tutorials#1523] [Q&A]
I was checking the implementation of densenet and I noticed that the class layers, defined as:
```python
  self.class_layers = nn.Sequential(
      OrderedDict(
          [
              ("relu", get_act_layer(name=act)),
              ("pool", avg_pool_type(1)),
              ("flatten", nn.Flatten(1)),
              ("out", nn.Linear(in_channels, out_channels)),
          ]
```

There is no softmax in the output, and I find this odd. Am I missing something?

**Comment by KumoLiu** [comment]
Hi @grudloff, I think you can add it by using `Activations`.
https://github.com/Project-MONAI/MONAI/blob/281cb0119c01eaa8e6c841880b91f92f45e8d7f7/monai/transforms/post/array.py#L59

Thanks!

**Comment by ericspod** [comment]
We have typically separated network outputs from activation so that you have the flexibility to do what you like with those raw values. Activation is done with postprocessing transforms like this: https://github.com/Project-MONAI/tutorials/blob/02a1664fdcb74e74133b49eab4089147286fb45d/3d_classification/ignite/densenet_training_dict.py#L124
